catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswSubscriptionEndpoint.java

```diff
@@ -0,0 +1,546 @@
+        // to registry
+        if (registeredSubscriptions.containsKey(subscriptionUuid)) {
+            LOGGER.debug("Delete existing subscription {} for re-creation", subscriptionUuid);
+            deleteSubscription(subscriptionUuid);
+        }
+        CswSubscription sub = createSubscription(request);
+        LOGGER.debug("Registering Subscription");
+        ServiceRegistration serviceRegistration =
+                context.registerService(Subscription.class.getName(), sub, props);
+
+        if (serviceRegistration != null) {
+            LOGGER.debug("Subscription registered with bundle ID = {} ",
+                    serviceRegistration.getReference()
+                            .getBundle()
+                            .getBundleId());
+            registeredSubscriptions.put(subscriptionUuid, serviceRegistration);
+            // Pass in client-provided subscriptionId vs. subscription UUID because
+            // the filter XML to be persisted for this subscription will be used to
+            // restore this subscription and should consist of the exact values the
+            // client originally provided.
+            if (persistSubscription) {
+                persistSubscription(sub, deliveryMethodUrl, subscriptionUuid);
+            }
+        } else {
+            LOGGER.debug("Subscription registration failed");
+        }
+
+        LOGGER.debug("EXITING: {}", methodName);
+        return subscriptionUuid;
+
+    }
+
+    private synchronized boolean deleteSubscription(String subscriptionId) throws CswException {
+        String methodName = "deleteSubscription";
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("ENTERING: {}", methodName);
+            LOGGER.debug("subscriptionId = {}", subscriptionId);
+        }
+
+        if (StringUtils.isEmpty(subscriptionId)) {
+            throw new CswException(
+                    "Unable to delete subscription because subscription ID is null or empty");
+        }
+
+        boolean status = false;
+
+        try {
+            LOGGER.debug("Removing (unregistering) subscription: {}", subscriptionId);
+            ServiceRegistration sr = (ServiceRegistration) registeredSubscriptions.remove(
+                    subscriptionId);
+            if (sr != null) {
+                sr.unregister();
+            } else {
+                LOGGER.debug("No ServiceRegistration found for subscription: {}", subscriptionId);
+            }
+
+            Configuration subscriptionConfig = getSubscriptionConfiguration(subscriptionId);
+            try {
+                if (subscriptionConfig != null) {
+                    LOGGER.debug("Deleting subscription for subscriptionId = {}", subscriptionId);
+                    subscriptionConfig.delete();
+
+                    // Subscription removal is only successful if able to remove from OSGi registry and
+                    // ConfigAdmin service
+                    if (sr != null) {
```

Comment:
If we keep the code the way it is, can we improve the error log to say that the bundle should be restarted and add `RuntimeException` to the catch block?

And just to confirm, the `CswSubscriptionConfigFactory` is the class that will reload the subscriptions in the map, right?